airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/cdc/MySqlDebeziumStateUtil.java

```diff
@@ -63,9 +64,9 @@ public boolean savedOffsetStillPresentOnServer(final JdbcDatabase database, fina
         LOGGER.info("Connector used GTIDs previously, but MySQL server does not know of any GTIDs or they are not enabled");
         return false;
       }
-      final GtidSet gtidSetFromSavedState = new GtidSet(savedState.gtidSet().get());
+      final MySqlGtidSet gtidSetFromSavedState = new MySqlGtidSet(savedState.gtidSet().get());
```

Comment:
I think they also have an implementation for [MariaDbGtidSet](https://github.com/debezium/debezium/blob/main/debezium-connector-mariadb/src/main/java/io/debezium/connector/mariadb/gtid/MariaDbGtidSet.java)

Can you test if this change would work on a maria db?

Reply:
yes, from my previous experience, mariadb and original MySQL are almost considered two different databases now with different storage engines. We probably shall really have another connector called mariadb that uses the GtidSet you mentioned above?

Reply:
Given the latest test on mariadb, I think we can probably continue using MySQL connector for mariadb ...